src/test/scala/impl/group1/MaximumNestingAnalysisTest.scala

```diff
@@ -0,0 +1,45 @@
+package org.tud.sse.metrics
+package impl.group1
+
+import testutils.AnalysisTestUtils
+
+import org.scalatest.{FlatSpec, Matchers}
+
+import java.io.File
+
+class MaximumNestingAnalysisTest extends FlatSpec with Matchers{
+
+  "The MaximumNestingAnalysis" must "calculate valid results for single JARs" in {
+
+    val fileToTest = new File(getClass.getResource("/group1/MaximumNestingCompanion-1.0-SNAPSHOT.jar").getPath)
+    val analysisToTest = new MaximumNestingAnalysis()
+
+    val appConfig = new ApplicationConfiguration(inputFilePath = "", treatFilesAsLibrary = true, outFileOption = None,
+      opalLoggingEnabled = false, batchModeEnabled = Some(false), excludedAnalysesNames = List(), includedAnalysesNames = List(),
+      excludeJreClasses = true, additionalClassesDir = None, loadAdditionalClassesAsInterface = false)
+
+    val result = AnalysisTestUtils.runSingleFileAnalysis(analysisToTest, fileToTest, appConfig, Map.empty[Symbol, Any])
+
+    assert(result.nonEmpty)
+
+    val metricResult = result.head
+
+    assert(metricResult.success)
+    assert(metricResult.metricValues.nonEmpty)
+    assert(metricResult.metricValues.exists(value => value.entityIdent.contains("class") && value.metricValue == 6))
+    assert(metricResult.metricValues.exists(value => value.entityIdent.contains("void start.<init>()") && value.metricValue == 0))
+    assert(metricResult.metricValues.exists(value => value.entityIdent.contains("method: int start.test7(int,int)") && value.metricValue == 6))
+    assert(metricResult.metricValues.exists(value => value.entityIdent.contains("method: int start.test6(int,int,int)") && value.metricValue == 2))
+    assert(metricResult.metricValues.exists(value => value.entityIdent.contains("method: int start.test5(int,int)") && value.metricValue == 6))
+    assert(metricResult.metricValues.exists(value => value.entityIdent.contains("method: int start.test4(int,int)") && value.metricValue == 3))
+    assert(metricResult.metricValues.exists(value => value.entityIdent.contains("method: int start.test3(int,int)") && value.metricValue == 2))
+    assert(metricResult.metricValues.exists(value => value.entityIdent.contains("method: int start.test2(int,int,int)") && value.metricValue == 3))
+    assert(metricResult.metricValues.exists(value => value.entityIdent.contains("method: int start.test1(int,int,int)") && value.metricValue == 2))
+    assert(metricResult.metricValues.exists(value => value.entityIdent.contains("method: int start.test8()") && value.metricValue == 3))
```

Comment:
Bei den Tests kommt "test8" 3-mal vor und die Testfälle "test9" und "test10" werden nicht gestestet, obwohl sie angelegt worden sind. Außerdem dürfte der Metrikwert bei "test8" doch 4 sein und nicht 3, oder?

Reply:
Eine weitere Anmerkung: Ich habe noch weiter etwas bei den Testfällen ausprobiert und interessanterweise ist Folgendes aufgetreten. Nachdem ich euren Source-Code für das Testprogramm (der in der txt-Datei steht) in Eclipse hineinkopiert habe und dort daraus noch einmal eine neue jar-File erstellt habe, habe ich diese Datei nochmal getestet. In diesem Fall funktionieren bei mir auch alle Tests und der Wert bei "test8" ist 4 (was meiner Meinung nach auch richtig ist). Meine Vermutung wäre jetzt, dass die jar-Datei nicht 100%-ig zu dem Source-Code passt bzw. dass beim Erstellen irgendwas schief gelaufen ist. Ist auf jeden Fall etwas merkwürdig.

Reply:
Wenn man von Hand zählt kommt wie erwartet 4 raus:
In Zeile 23 soll gegebenenfalls nach Zeile 62 gesprungen werden
In Zeile 62 wird stets nach Zeile 9 gesprungen
Wenn ich im IntelliJ-Debugger die Metrik durchsteppe zeigt er mir für Zeile nicht if_"icmpge  62" an
sondern "if_icmpge 23-14= 9" an
Es werden also unerwarteterweise 2 Instruktionen zusammengefaßt. Durch den unerwarteten Rücksprung verrechnet sich die Metrik
```
int test8();
    Code:
       0: sipush        255
       3: istore_1
       4: bipush        20
       6: istore_2
       7: iconst_1
       8: istore_3
       9: iload_3
      10: ifeq          65
      13: iload_2
      14: iload_1
      15: iadd
      16: istore_2
      17: iconst_0
      18: istore        4
      20: iload         4
      22: iload_1
      23: if_icmpge     62
      26: bipush        10
      28: istore        5
      30: iload         5
      32: iload_1
      33: iload_1
      34: imul
      35: if_icmpge     56
      38: iload         5
      40: iload_1
      41: iload_1
      42: imul
      43: iconst_1
      44: isub
      45: if_icmpne     50
      48: iconst_0
      49: istore_3
      50: iinc          5, 1
      53: goto          30
      56: iinc          4, 1
      59: goto          20
      62: goto          9
      65: iload_2
      66: ireturn



```